### PR TITLE
bf16_optimizer + immediate_grad_update: skip grad acc on backward

### PR DESCRIPTION
**Pull request by nelyahu**
when using immediate_grad_update, gradient accumulation is performed using hook during the backward of each module. thus the call to update_hp_grads at the end of the backward has to be skipped to avoid additional overhead. although no accuracy issue is expected as gradients are cleared after the hook.

### CODE REVIEW

File: deepspeed/runtime/bf16_optimizer.py [hunk 1]
```diff
@@ -338,7 +338,7 @@ def backward(self, loss, update_hp_grads=True, clear_lp_grads=False, **bwd_kwarg
         self.clear_lp_grads()
         loss.backward(**bwd_kwargs)
 
-        if update_hp_grads:
+        if not self.immediate_grad_update and update_hp_grads:
```
[the remark this answers is not in the document]
Comment (nelyahu) [reply deepspeed/runtime/bf16_optimizer.py:341]: @tjruwase correct missed it, we have this redundancy on our code base. will fix locally.
